include/boost/mysql/impl/internal/protocol/deserialization.hpp

```diff
@@ -290,7 +292,13 @@ boost::mysql::error_code boost::mysql::detail::deserialize_ok_packet(
 
     if (ctx.enough_size(1))  // message is optional, may be omitted
     {
-        err = pack.info.deserialize(ctx);
+        if (flavor == db_flavor::clickhouse) {
```

Comment:
I think you can avoid the variant altogether - just create and deserialize the string_lenenc or string_eof depending on the case. Since this lib supports back to cpp11, unfortunately you can't use generic lambdas, so I think this is easier.